kotlin/tests/src/test/kotlin/com/squareup/moshi/kotlin/DefaultConstructorTest.kt

```diff
@@ -81,6 +81,13 @@ data class TestClass(
     val dynamicInlineOptional: Int = createInlineInt()
 )
 
+// Regression test for https://github.com/square/moshi/issues/905
+// Just needs to compile
+@JsonClass(generateAdapter = true)
+data class GenericTestClassWithDefaults(
```

Comment:
Missed `<T>`

Reply:
d'oh. And doing this locally reveals it's not actually fixed. Good catch!

Will wait until #908 is merged because the code gen changes there will make this easier